src/modules/sensors/sensors.cpp

```diff
@@ -490,71 +501,102 @@ Sensors::adc_poll(struct sensor_combined_s &raw)
 								 ORB_PRIO_DEFAULT);
 					}
 
+#endif
+
+#ifdef ADC_AIRSPEED_VOLTAGE_CHANNEL
+#ifndef ADC_SONAR_VOLTAGE_CHANNEL
+
 				} else
 #endif
-				{
-					for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
+#endif
 
-						/* Once we have subscriptions, Do this once for the lowest (highest priority
-						 * supply on power controller) that is valid.
+#ifdef ADC_SONAR_VOLTAGE_CHANNEL
+#ifdef ADC_AIRSPEED_VOLTAGE_CHANNEL
+
+				} else if (ADC_SONAR_VOLTAGE_CHANNEL == buf_adc[i].am_channel) {
+#else
+
+					if (ADC_SONAR_VOLTAGE_CHANNEL == buf_adc[i].am_channel) {
+#endif
+				float voltage_scaling = 2.04f;
+				float distance = buf_adc[i].am_data * 0.001f * voltage_scaling;
+
+				struct distance_sensor_s distance_report;
+				distance_report.timestamp = t;
+				distance_report.type = distance_sensor_s::MAV_DISTANCE_SENSOR_ULTRASOUND;
+				distance_report.orientation = 8;
```

Comment:
I'm trying to set the distance sensor type as a parameter and I would like to make it so user can choose the type from drop down menu (e.g. like `SYS_MC_EST_GROUP`). 
This is what has been generated for me in `parameters.xml` file:
```
<parameter default="0" name="DIST_SEN_TYPE" type="INT32">
      <short_desc>Distance sensor type</short_desc>
      <min>0</min>
      <max>3</max>
      <reboot_required>false</reboot_required>
      <scope>modules/sensors</scope>
      <values>
        <value code="1">ULTRASOUND</value>
        <value code="0">LASER</value>
        <value code="3">RADAR</value>
        <value code="2">INFRARED</value>
      </values>
    </parameter>
```
But I can't see the dropdown menu. I can only set the value manually. I also did `(void)param_find("DIST_SEN_TYPE");` according to this comment: "These are parameters for which QGroundControl always expects to be returned in a list request." in `modules/sensors/parameters.cpp` file. 

Am I missing something else?